Accessing and setting the site config

Starbugstone opened the pull request:
pushing latest updates to dev

Starbugstone committed:
Config retreval

Creating the configmodel for retreaving all the config elements
displaying via the admin config controller and using admin/config.twig for the render

Starbugstone committed:
Merge branch 'dev' into Accessing-and-setting-the-site-config

Starbugstone committed:
updating our module charger

We can now implement modules in 'namespace'\Modules and app\Modules as well as from Core.
 they are prioritised by that order to enable surcharging methods if needed.

Codacy will probably complain about "else" not necessary but fuck it !! this is the BEST way to do the check

Starbugstone committed:
adding a check if in dev mode and multiple modules declared

We could have multiple modules declared whith app or namespace modules over-charging the core and app modules.
We now have an error message to warn if in dev mode.
in prod mode, we should know what we are doing so no message.
And meny if's because of codacy since it hates the 'else'

Starbugstone committed:
extra tests on the module loader

Adding some extra checks on the module loader

Starbugstone committed:
making our module loader easier to read and updating the dev panel

Adding extra functions to simplify code and make more robust.
Also putting the called module in the dev panel rather than throwing alerts.
Cleaner

Starbugstone committed:
Removing the $_SESSIOn superglobal access

Moved the superglobal access from the controller to the session class to keep things cleaner.

Oipnet commented on App/Models/ConfigModel.php:
```diff
@@ -0,0 +1,41 @@
+    {
+        $returnData = [];
+        //Grabbing the configs class
+        $sql = 'select * from configs_class order by class';
```
why use select * if you want only classname

Starbugstone replied:
because the id is also needed. Bind in the foreach on line 35.
I could probably find a better sql query rather than querying for each class

Starbugstone committed:
upper and lowercase inconsistance

My Linux box doesn't like !!

Starbugstone committed:
updated the get config model

Removed the multiple sql queries and grab all the data with an inner join.

Also added the fetch() and fetchAll() to the code model to avoid all the $this->stmt->fetchall() stuff.

Starbugstone commented:
@Oipnet Ok, select * problem fixed. Only one call to the DB \o/

EDIT : Just realised that I will have a problem with prefixed tables with the queries

Starbugstone commented:
this part isn't finished. The pull request was an error.
Closing for now
